libs/@local/harpc/server/examples/account.rs

```diff
@@ -115,146 +120,78 @@ impl Procedure for CreateAccount {
     const ID: <Self::Subsystem as Subsystem>::ProcedureId = AccountProcedureId::CreateAccount;
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, thiserror::Error)]
-enum AccountError {
-    #[error("unable to establish connection to server")]
-    Connection,
-    #[error("unable to encode request")]
-    Encode,
-    #[error("unable to decode response")]
-    Decode,
-    #[error("expected at least a single response")]
-    ExpectedResponse,
-}
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, derive_more::Display, derive_more::Error)]
+#[display("unable to fullfil account request")]
+pub struct AccountError;
```

Comment:
Let's do that for any error we touch:
```suggestion
#[must_use]
pub struct AccountError;
```

Reply:
Also, let's try to use `derive_more` more often (as you did in this PR)

Reply:
see [575d32b](https://github.com/hashintel/hash/pull/5608/commits/575d32b4973e2b0570429e8a7880f917590c633e)